app/common/directives/embed-only.directive.js

```diff
@@ -11,8 +11,9 @@ function EmbedOnlyDirective() {
 EmbedOnlyController.$inject = ['$scope', '$element', '$attrs', '$rootScope', '_', '$window'];
 function EmbedOnlyController($scope, $element, $attrs, $rootScope, _, $window) {
     var globalEmbed = ($window.self !== $window.top) ? true : false;
-    if (globalEmbed) {
-        $rootScope.setLayout('layout-embed');
+    if (globalEmbed && $rootScope.globalLayout.indexOf('layout-embed') < 0) {
```

Comment:
.. or actually if we just always use the `layout-class` directive rather than calling `setLayout` won't this take care of itself?